### PR TITLE
Added JSONRPCClient testing

### PR DESCRIPTION
**Pull request by afragapane**
### Description of the Change
Title.

### Applicable Issues
#45

### CODE REVIEW

File: packages/neo-one-client-core/src/__data__/transactions.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/* @flow */
+import {
+  ASSET_TYPE,
+  RegisterTransaction,
+  ScriptBuilder,
+  common,
+  crypto,
+} from '../';
```
Comment (dicarlo2) [inline packages/neo-one-client-core/src/__data__/transactions.js]: Move this to `neo-one-client` and change this line to `} from '@neo-one-client-core'`

File: packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js [hunk 2]
```diff
@@ -0,0 +1,161 @@
+/* @flow */
+import BigNumber from 'bignumber.js';
+
+import { keys, transactions } from '@neo-one/client-core/src/__data__';
+
+import JSONRPCClient from '../../../provider/neoone/JSONRPCClient';
+import { RelayTransactionError } from '../../../errors';
+
+jest.mock('isomorphic-fetch');
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js]: I think this can be removed, we're just forwarding calls to the underlying provider and no `fetch` is happening.

File: packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js [hunk 3]
```diff
@@ -0,0 +1,161 @@
+/* @flow */
+import BigNumber from 'bignumber.js';
+
+import { keys, transactions } from '@neo-one/client-core/src/__data__';
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js]: Yea, typically not good to assume the structure of an external module. Instead, let's copy the `keys` file to the `__data__` directory inside `neo-one-client`. It's definitely duplication, but in this case it's a bit cleaner than trying to import like this.

File: packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      method: 'testInvocation',
+      args: [Buffer.alloc(1, 0)],
+    },
+    {
+      method: 'getTransactionReceipt',
+      args: [keys[0].scriptHash],
+    },
+    {
+      method: 'getTransactionReceipt',
+      args: [keys[0].scriptHash],
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js]: Should also test the case where options is non-null (so `watchTimeout` is defined in the snapshot, see line 183 in the snapshot file)

File: packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+      method: 'getAsset',
+      args: [transactions.register.hash],
+    },
+    {
+      method: 'getBlock',
+      args: [transactions.register.hash],
+    },
+    {
+      method: 'getBlock',
+      args: [0],
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/JSONRPCClient.test.js]: Same here as `getTransactionReceipt`, add a case where `options` is non-null so `watchTimeout` is defined in the snapshot.

File: packages/neo-one-client/src/__data__/index.js [hunk 6]
```diff
@@ -0,0 +1,4 @@
+/* @flow */
+// eslint-disable-next-line
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__data__/index.js]: remove

File: packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js [hunk 7]
```diff
@@ -0,0 +1,810 @@
...
+import { AsyncIterableX } from 'ix/asynciterable/asynciterablex';
+import BigNumber from 'bignumber.js';
+
+import { toArray } from 'ix/asynciterable/toarray';
+
+import NEOONEDataProvider from '../../../provider/neoone/NEOONEDataProvider';
+
+import { keys, transactions } from '../../../__data__';
+
+function genExpectedInvocationResult(options: Object) {
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js]: In some languages, e.g. php/hack, prefixing a method with `gen` indicates it's an async method. We don't follow that convention (and generally it's not done in JavaScript), but it'd be best to avoid using that prefix to avoid confusion. Instead, perhaps prefix with `create`. Same with all the other methods.

File: packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js [hunk 8]
```diff
@@ -0,0 +1,810 @@
...
+        expected: genExpectedTransaction('PublishTransaction', {
+          contract: genExpectedContract({}),
+        }),
+        transactionJSON: genTransactionJSON('PublishTransaction', {
+          contract: genContractJSON({}),
+        }),
+      },
+      {
+        expected: genExpectedTransaction('RegisterTransaction', {
+          asset: (({ type, name, amount, precision, owner, admin }) => ({
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js]: Extract this logic out into a common function, like `createRegisterExpectedAsset`.

Comment (dicarlo2) [reply packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js]: Also, as a small nit, it took me a second to figure out what was going on here - creating an anonymous function just to extract some of the properties is not a very common pattern.

File: packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js [hunk 9]
```diff
@@ -0,0 +1,810 @@
...
+    );
+
+    const result = provider.getTransactionReceipt(transactions.register.hash);
+    expect(result).resolves.toEqual(expected);
+  });
+
+  test('getInvocationData', async () => {
+    const expected = genExpectedInvocationData({});
+
+    // JSONHelper.readFixed8 = jest.fn(() => 0);
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js]: Remove

File: packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js [hunk 10]
```diff
@@ -0,0 +1,810 @@
...
+    // $FlowFixMe
+    provider._client.getNetworkSettings = jest.fn(() =>
+      Promise.resolve({ issueGASFee: '0' }),
+    );
+
+    const result = provider.getNetworkSettings();
+    expect(result).resolves.toEqual(expected);
+  });
+
+  test('_ConvertConfirmedTransaction throws null data error', async () => {
```
Comment (dicarlo2) [inline packages/neo-one-client/src/__tests__/provider/neoone/NEOONEDataProvider.test.js]: nit: `_convertConfirmedTransaction`
